pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -409,6 +409,51 @@ func (cr *PerconaXtraDBCluster) VersionLessThan120() bool {
 	return currentVersion.LessThan(checkVersion)
 }
 
+func (cr *PerconaXtraDBCluster) VVersionGreaterOrEqual130() bool {
+	apiVersion := cr.APIVersion
+	if lastCR, ok := cr.Annotations["kubectl.kubernetes.io/last-applied-configuration"]; ok {
+		var newCR PerconaXtraDBCluster
+		err := json.Unmarshal([]byte(lastCR), &newCR)
+		if err != nil {
+			return false
+		}
+		apiVersion = newCR.APIVersion
+	}
+	crVersion := strings.Replace(strings.TrimLeft(apiVersion, "pxc.percona.com/v"), "-", ".", -1)
+	checkVersion, err := v.NewVersion("1.3.0")
+	if err != nil {
+		return false
+	}
+	currentVersion, err := v.NewVersion(crVersion)
+	if err != nil {
+		return false
+	}
+	return currentVersion.GreaterThanOrEqual(checkVersion)
+}
+
+// CompareVersion compares given version to current version. Returns -1, 0, or 1 if given version is smaller, equal, or larger than the current version, respectively.
+func (cr *PerconaXtraDBCluster) CompareVersion(version string) (int, error) {
+	apiVersion := cr.APIVersion
+	if lastCR, ok := cr.Annotations["kubectl.kubernetes.io/last-applied-configuration"]; ok {
+		var newCR PerconaXtraDBCluster
+		err := json.Unmarshal([]byte(lastCR), &newCR)
+		if err != nil {
+			return 0, err
+		}
+		apiVersion = newCR.APIVersion
+	}
```

Comment:
I just realised that I don't understand the logic behind taking the version from the "kubectl.kubernetes.io/last-applied-configuration" annotation. We already have the latest config in `cr`, innit?